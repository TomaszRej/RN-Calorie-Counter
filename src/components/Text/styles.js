import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  textColor: {color: 'black'},
  defaultSize: {fontSize: 12},
  h1: {fontSize: 17},
  h2: {fontSize: 16},
  h3: {fontSize: 15},
  medium: {fontSize:13 },
  small: {fontSize: 10},
});
