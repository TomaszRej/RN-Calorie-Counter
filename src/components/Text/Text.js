/**
 * @flow
 */

import * as React from 'react';
import {Text as RNText} from 'react-native';
import {FontWeight} from 'src/utils/FontWeight';
import {styles} from './styles';

type Props = {
  style?: RNText.propTypes.style,
  weight:
    | 'thin'
    | 'ultraLight'
    | 'light'
    | 'regular'
    | 'medium'
    | 'semiBold'
    | 'bold'
    | 'extraBold'
    | 'heavy'
    | 'black',
  fontFamily: 'Lato',
  size: 'h1' | 'h2' | 'h3' | 'default' | 'medium' | 'small',
  textType?: 'italic' | null,
  // eslint-disable-next-line flowtype/no-weak-types
  children?: string | React$Node,
  numberOfLines?: number,
  color?: string,
  align: 'left' | 'right' | 'center',
};
export const Text = ({
  style,
  weight,
  fontFamily,
  size,
  textType,
  children,
  numberOfLines,
  color,
  align,
}: Props): React.Node => {
  let fontSizeStyle;

  switch (size) {
    case 'small': {
      fontSizeStyle = styles.small;
      break;
    }
    case 'medium': {
      fontSizeStyle = styles.medium;
      break;
    }

    case 'h1': {
      fontSizeStyle = styles.h1;
      break;
    }
    case 'h2': {
      fontSizeStyle = styles.h2;
      break;
    }
    case 'h3': {
      fontSizeStyle = styles.h3;
      break;
    }
    default: {
      fontSizeStyle = styles.defaultSize;
      break;
    }
  }

  return (
    <RNText
      numberOfLines={numberOfLines}
      style={[
        {textAlign: align},
        FontWeight(fontFamily, weight, textType),
        fontSizeStyle,
        styles.textColor,
        style,
        color && {color},
      ]}
    >
      {children}
    </RNText>
  );
};

Text.defaultProps = {
  weight: 'regular',
  size: 'default',
  textType: null,
  fontFamily: 'Lato',
  align: 'left',
};
