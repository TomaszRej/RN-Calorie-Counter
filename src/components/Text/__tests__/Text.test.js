import React from 'react';
import {View} from 'react-native';
import {shallow} from 'enzyme';
import {Text} from '../Text';

describe('Text', () => {
  describe('Rendering', () => {
    it('should match to snapshot - Default', () => {
      const component = shallow(<Text>Test</Text>);
      expect(component).toMatchSnapshot();
    });
    it('should match to snapshot - Italics', () => {
      const component = shallow(<Text textType="italic">Test</Text>);
      expect(component).toMatchSnapshot();
    });
    it('should match to snapshot - align right', () => {
      const component = shallow(<Text align="right">Test</Text>);
      expect(component).toMatchSnapshot();
    });
    it('should match to snapshot - Default Bold', () => {
      const component = shallow(<Text weight="bold">Test</Text>);
      expect(component).toMatchSnapshot();
    });
    it('should match to snapshot - Default Medium', () => {
      const component = shallow(<Text weight="medium">Test</Text>);
      expect(component).toMatchSnapshot();
    });
    it('should match to snapshot - Default Thin', () => {
      const component = shallow(<Text weight="thin">Test</Text>);
      expect(component).toMatchSnapshot();
    });

    it('should match to snapshot - H1', () => {
      const component = shallow(<Text size="h1">Test</Text>);
      expect(component).toMatchSnapshot();
    });
    it('should match to snapshot - H2', () => {
      const component = shallow(<Text type="h2">Test</Text>);
      expect(component).toMatchSnapshot();
    });
    it('should match to snapshot - H3', () => {
      const component = shallow(<Text type="h3">Test</Text>);
      expect(component).toMatchSnapshot();
    });
    it('should match to snapshot - Medium', () => {
      const component = shallow(<Text type="medium">Test</Text>);
      expect(component).toMatchSnapshot();
    });
    it('should match to snapshot - Small', () => {
      const component = shallow(<Text type="small">Test</Text>);
      expect(component).toMatchSnapshot();
    });
  });

  describe('Chceck Props', () => {
    describe('type', () => {
      let props;
      let component;
      const weight = 'bold';
      const type = 'h3';
      const text = 'Test';
      const style = {color: 'red'};

      beforeEach(() => {
        component = shallow(
          <View>
            <Text style={style} weight={weight} type={type}>
              {text}
            </Text>
          </View>,
        );
        props = component.props().children.props;
        jest.clearAllMocks();
      });

      it('weight should be bold', () => {
        expect(props.weight).toEqual(weight);
      });
      it('type should be small', () => {
        expect(props.type).toEqual(type);
      });
      it('style ', () => {
        expect(props.style).toEqual(style);
      });
      it('ext should be "Test" ', () => {
        expect(props.children).toEqual(text);
      });
    });
  });
});
