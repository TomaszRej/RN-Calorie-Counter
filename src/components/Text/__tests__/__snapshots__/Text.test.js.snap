// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Rendering should match to snapshot - Default 1`] = `
<Text
  style={
    Object {
      "color": "#3E3F3F",
      "fontFamily": "Lato-Regular",
      "fontSize": 12,
      "fontStyle": null,
      "fontWeight": "400",
      "textAlign": "left",
    }
  }
>
  Test
</Text>
`;

exports[`Text Rendering should match to snapshot - Default Bold 1`] = `
<Text
  style={
    Object {
      "color": "#3E3F3F",
      "fontFamily": "Lato-Bold",
      "fontSize": 12,
      "fontStyle": null,
      "fontWeight": "700",
      "textAlign": "left",
    }
  }
>
  Test
</Text>
`;

exports[`Text Rendering should match to snapshot - Default Medium 1`] = `
<Text
  style={
    Object {
      "color": "#3E3F3F",
      "fontFamily": "Lato-Medium",
      "fontSize": 12,
      "fontStyle": null,
      "fontWeight": "500",
      "textAlign": "left",
    }
  }
>
  Test
</Text>
`;

exports[`Text Rendering should match to snapshot - Default Thin 1`] = `
<Text
  style={
    Object {
      "color": "#3E3F3F",
      "fontFamily": "Lato-Thin",
      "fontSize": 12,
      "fontStyle": null,
      "fontWeight": "100",
      "textAlign": "left",
    }
  }
>
  Test
</Text>
`;

exports[`Text Rendering should match to snapshot - H1 1`] = `
<Text
  style={
    Object {
      "color": "#3E3F3F",
      "fontFamily": "Lato-Regular",
      "fontSize": 20,
      "fontStyle": null,
      "fontWeight": "400",
      "textAlign": "left",
    }
  }
>
  Test
</Text>
`;

exports[`Text Rendering should match to snapshot - H2 1`] = `
<Text
  style={
    Object {
      "color": "#3E3F3F",
      "fontFamily": "Lato-Regular",
      "fontSize": 12,
      "fontStyle": null,
      "fontWeight": "400",
      "textAlign": "left",
    }
  }
>
  Test
</Text>
`;

exports[`Text Rendering should match to snapshot - H3 1`] = `
<Text
  style={
    Object {
      "color": "#3E3F3F",
      "fontFamily": "Lato-Regular",
      "fontSize": 12,
      "fontStyle": null,
      "fontWeight": "400",
      "textAlign": "left",
    }
  }
>
  Test
</Text>
`;

exports[`Text Rendering should match to snapshot - Italics 1`] = `
<Text
  style={
    Object {
      "color": "#3E3F3F",
      "fontFamily": "Lato-RegularItalic",
      "fontSize": 12,
      "fontStyle": null,
      "fontWeight": "400",
      "textAlign": "left",
    }
  }
>
  Test
</Text>
`;

exports[`Text Rendering should match to snapshot - Medium 1`] = `
<Text
  style={
    Object {
      "color": "#3E3F3F",
      "fontFamily": "Lato-Regular",
      "fontSize": 12,
      "fontStyle": null,
      "fontWeight": "400",
      "textAlign": "left",
    }
  }
>
  Test
</Text>
`;

exports[`Text Rendering should match to snapshot - Small 1`] = `
<Text
  style={
    Object {
      "color": "#3E3F3F",
      "fontFamily": "Lato-Regular",
      "fontSize": 12,
      "fontStyle": null,
      "fontWeight": "400",
      "textAlign": "left",
    }
  }
>
  Test
</Text>
`;

exports[`Text Rendering should match to snapshot - align right 1`] = `
<Text
  style={
    Object {
      "color": "#3E3F3F",
      "fontFamily": "Lato-Regular",
      "fontSize": 12,
      "fontStyle": null,
      "fontWeight": "400",
      "textAlign": "right",
    }
  }
>
  Test
</Text>
`;
